### ISSUE TITLE
When using podman kube play, allow to specify Dockerfile/build folder location

### ISSUE THREAD

**Issue by BernhardPosselt** [containers/podman#12485]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**

podman play kube allows you to [build a container from a local container file](https://docs.podman.io/en/latest/markdown/podman-play-kube.1.html#description) by using a subdirectory as the image value.

This is fine if you set up a new project, but if you retroactively want to enable in your project, it will require a giant commit, possibly moving thousands of files into a subfolder, breaking peoples ide setups and potentially builds.

The main issue here is that Dockerfiles/Containerfiles can not reference directories that are above their current folder; the following line can not be built:

```
COPY ../directory /server/directory
```

This only leaves you with moving everything to a sub directory

docker-compose solves this by being able to [specify the build directory and Dockerfile location](https://docs.docker.com/compose/compose-file/compose-file-v3/#build) whereas the following is not possible using the kube configuration:

```yaml
spec:
  containers:
  - image: "."
    name: server
```

**Comment by rhatdan** [comment]
SGTM, But the Dockerfile and context directory are per image, I believe.  How should it be handled if there is more then one image?

**Comment by rhatdan** [comment]
@baude WDYT?

**Comment by BernhardPosselt** [comment]
Yep, they are per image. I'm currently trying to learn podman/k8s and migrate off Docker so I'm not 100% sure but the [yaml spec doesn't look like there can ever be more than 1 image per container](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#Container)

**Comment by rhatdan** [comment]
Right, but what if you have two containers in a Pod? Then  you would have two images to build potentially.

**Comment by baude** [comment]
yeah this gets kind of ugly quickly, which is why I asked @BernhardPosselt to make the issue.  could we allow the context dir to be set in addition to the subdir.  it would have to then be for all images.

the other idea i was chewing on is some sort of k8s annotations ... but we cannoot auto-gen those. it would all be manual.   k8s would ignore it

**Comment by BernhardPosselt** [comment]
Wouldn't it be possible to attach that build information on the container level itself? Otherwise I actually wouldn't mind going for something like annotations that define per image where my Dockerfile and context directory are.

As for the use case: looked into it a bit more but then realized that I can't even move it around in the repository because I would break the deployment scripts (which I can't edit, don't ask why xD) as well. 

So basically I really need a way to be able to build from the same location where the pod yaml is in, which is currently only possible using a docker-compose.yml.

**Comment by rhatdan** [comment]
Currently we look for the subdirs with the image name.  We could point to a different directory, which also had the image named subdirs.  But that is not really what you want.

**Comment by flouthoc** [comment]
> Yep, they are per image. I'm currently trying to learn podman/k8s and migrate off Docker so I'm not 100% sure but the [yaml spec doesn't look like there can ever be more than 1 image per container](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#Container)

I am pretty sure kubernetes does not supports this but this could be a podman only [RFE].

Developers use tools like `skaffold` (https://skaffold.dev/docs/) as client instead of `kubectl` with `kubernetes` which performs building of images https://skaffold.dev/docs/pipeline-stages/builders/` locally.

If this is a podman only `RFE` then I'd agree with @baude pod annotations looks like a really good way to implement this

I personally think best approach to solve this problem without tampering with `k8s` api struct would be.
* define a small build `protocol/specification` using pod annotations : Declarative solution
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: mypod
  annotations:
    podman-build: "{image: <imagename>, context: </path/context>, containerfile: <path/to/containerfile>}"
    podman-build: "{image: <imagename2>, context: </path/context2>, containerfile: <path/to/containerfile2>}"
```

* Second would be not declarative with similar format in `podman play kube` cli

**Comment by vrothberg** [comment]
Friendly ping.

@BernhardPosselt would the build annotations as outlined above satisfy your needs?

**Comment by BernhardPosselt** [comment]
Yes.

**Comment by flouthoc** [comment]
@vrothberg Sure I'd love to.

**Comment by reavessm** [comment]
Did this feature get implemented?  The PR listed here doesn't seem to allow specific builds per image, only one `context-dir` for the entire file

**Comment by rhatdan** [comment]
Open a separate Issue for that.
